src/extras/render-passes/render-pass-depth-aware-blur.js

```diff
@@ -1,93 +1,197 @@
-            }
+        this._type = value;
 
-            // TODO: weights of 1 are used for all samples. Test with gaussian weights
-            void main() {
+        this.updateWeightCoefs();
+    }
 
-                // handle the center pixel separately because it doesn't participate in bilateral filtering
-                float depth = -getLinearScreenDepth(uv0);
-                float totalWeight = 1.0;
-                float color = texture2D(sourceTexture, uv0 ).r;
-                float sum = color * totalWeight;
+    get type() {
+        return this._type;
+    }
 
-                for (int x = -filterSize; x <= filterSize; x++) {
-                   for (int y = -filterSize; y < filterSize; y++) {
-                       float weight = 1.0;
-                       vec2 offset = vec2(x,y) * sourceInvResolution;
-                       tap(sum, totalWeight, weight, depth, uv0 + offset);
-                   }
-                }
+    set kernelSize(value) {
+        if (value === this._kernelSize) {
+            return;
+        }
 
-                float ao = sum / totalWeight;
+        this._kernelSize = Math.max(1, Math.floor(value));
 
-                // simple dithering helps a lot (assumes 8 bits target)
-                // this is most useful with high quality/large blurs
-                // ao += ((random(gl_FragCoord.xy) - 0.5) / 255.0);
+        this.updateShader();
 
-                gl_FragColor.r = ao;
-            }
-        `
-        );
+        this.updateWeightCoefs();
+    }
 
-        var scope = this.device.scope;
-        this.sourceTextureId = scope.resolve('sourceTexture');
-        this.sourceInvResolutionId = scope.resolve('sourceInvResolution');
-        this.sourceInvResolutionValue = new Float32Array(2);
-        this.filterSizeId = scope.resolve('filterSize');
+    /**
+     * @type {number}
+     */
+    get kernelSize() {
+        return this._kernelSize;
     }
 
-    execute() {
+    /**
+     * Sets the direction of the blur.
+     * @param {Vec2|Float32Array|Float64Array} value - The direction of the blur.
+     */
+    set direction(value) {
+        if (value instanceof Vec2) {
+            this._direction.set([value.x, value.y]);
+        } else {
+            this._direction.set(value);
+        }
+    }
 
-        this.filterSizeId.setValue(4);
-        this.sourceTextureId.setValue(this.sourceTexture);
+    /**
+     * @type {Float32Array}
+     */
+    get direction() {
+        return this._direction;
+    }
 
-        const { width, height } = this.sourceTexture;
-        this.sourceInvResolutionValue[0] = 1.0 / width;
-        this.sourceInvResolutionValue[1] = 1.0 / height;
-        this.sourceInvResolutionId.setValue(this.sourceInvResolutionValue);
+    updateShader() {
+        const shaderName = `BlurShader-kernelsize-${this.kernelSize}`;
 
-        super.execute();
+        const defines = `#define KERNEL_SIZE ${this.kernelSize}\n`;
+
+        // CHECK: should we destroy the shader?
```

Comment:
shaders are expensive to compile, so we don't destroy them, to make it faster when the shader is needed again

Reply:
Then I will remove the comment.